### ISSUE TITLE
Headless Chrome Lighthouse Unable to connect to Chrome

### ISSUE THREAD

**Issue by joelhsmith** [microsoft/WSL#4982]
* Your Windows build number:  *

Microsoft Windows [Version 10.0.19041.153]
Ubuntu 18.04

* What you're doing and what's happening: 

In WSL 2, I am trying to run a Google Chrome Lighthouse audit on a website in --headless mode.

```bash
$ npm --version
6.13.4
$lighthouse --version
5.6.0
$ lighthouse https://pets.com --chrome-flags="--headless --no-sandbox" --output=json --output-path=foo.json
  ChromeLauncher Waiting for browser. +0ms
  ChromeLauncher Waiting for browser... +1ms
  ChromeLauncher Waiting for browser..... +507ms
  ChromeLauncher Waiting for browser....... +503ms
blau blau blau
  ChromeLauncher Waiting for browser..................................................................................................... +502ms
  ChromeLauncher Waiting for browser....................................................................................................... +502ms
  ChromeLauncher:error connect ECONNREFUSED 127.0.0.1:36695 +2ms
  ChromeLauncher:error Logging contents of /mnt/c/Users/Joel/AppData/Local/lighthouse.26915742/chrome-err.log +25ms
  ChromeLauncher:error 
  ChromeLauncher:error DevTools listening on ws://127.0.0.1:36695/devtools/browser/80d94a0b-7734-4a51-808b-74f584542571
  ChromeLauncher:error  +0ms
Unable to connect to Chrome
```

* For WSL launch issues, please collect detailed logs

Total contents of chrome-err.log file above:

```
DevTools listening on ws://127.0.0.1:42073/devtools/browser/43fd7c2d-8251-4316-a68b-b46c28e9f735
```

I have tried to get the [detailed logs](https://github.com/Microsoft/WSL/blob/master/CONTRIBUTING.md#8-detailed-logs) from WSL, as requested, but I don't see those files in the location specified, sorry. 

* What's wrong / what should be happening instead:

It should result in a Lighthouse report called foo.json.  However, it looks like it can't launch

Why is the connection refused? Why can't it connect to Chrome?

I have exhausted all Chrome Lighthouse issues queues, and all four similar issues in here with no luck. So I am thinking it could be WSL 2 Ubuntu related.  Any troubleshooting advice would be most welcome.

**Comment by sirredbeard** [comment]
Off the top of my head, it looks like it's launching and failing to bind to 127.0.0.1:36695.

First make sure you have permissions to bind :36695. 

I also think it might be a networking issue.

You may need to bind your your VM's IP address (`$ ip addr show eth0 | grep -oP '(?<=inet\s)\d+(\.\d+){3}' `) or 0.0.0.0, see here:

https://docs.microsoft.com/en-us/windows/wsl/wsl2-ux-changes#other-networking-considerations

**Comment by joelhsmith** [comment]
Thank @sirredbeard for the link and the advice.  I don't quite understand what it means, tho. I love WSL 2, but I still new to all this.  Apologies if these are dumb super noob questions.

I'm not sure how to check if I have permission to bind :36695 .  How do I test that?

I'm not familiar with the way WSL 2 networking works. The result from the command you suggested is 172.26.191.31. FWIW, I am able to ping that and 0.0.0.0 successfully.  What do I do from there?

Any additional advice would be very welcome.

**Comment by sirredbeard** [comment]
On WSL2 your WSL instance has it's own internal IP address and is using what is called bridged networking, the networking on WSL is bridged to your Windows networking.

This is different from WSL1 where your WSL instance had the exact same IP address as Windows and used the same network interface.

This can result in some networking issues.

I would suggest you look at the docs for Google Chrome lighthouse and see if it gives you a command line option to set your IP address when launching.

If so, I would then experiment with their 172... or 0... and see if that works for you.

To test if you can bind ports you could use a little Python and experiment:

```
import socket, errno

s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)

try:
    s.bind(("127.0.0.1", 5555))
    print("Port works!")
except socket.error as e:
    if e.errno == errno.EADDRINUSE:
        print("Port is already in use")
    else:
        # something else raised the socket.error exception
        print(e)

s.close()

```

**Comment by joelhsmith** [comment]
This seems to be fixed on its own with recent releases of WSL.

**Comment by kayarre** [comment]
I am running into this same issue with current versions of wsl2.

**Comment by noelchapkowski** [comment]
This is still occurring and not fixed.

**Comment by Danarvelini** [comment]
Still occurring. Is there any fix?

**Comment by csaba-kovacs** [comment]
Im facing the same issue, unabel to figure out.
![Screenshot 2023-02-20 at 15 56 32](https://user-images.githubusercontent.com/88077449/220127063-3a81c2f2-1470-4994-8983-fbfa4d22c366.png)

**Comment by micktg** [comment]
I hade also no luck

**Comment by csaba-kovacs** [comment]
i reworked it with instructions from here https://web.dev/lighthouse-ci/ if it helps anyone.

**Comment by micktg** [comment]
> i reworked it with instructions from here https://web.dev/lighthouse-ci/ if it helps anyone.

can you summarize what you did to make it work?

**Comment by micktg** [comment]
Thanks, I will give it a try

**Comment by mugwhump** [comment]
Having the same issue, tried it with LHCI too. 
There's no reason LHCI would change things; it runs lighthouse the same way and still needs to run an instance of chrome. The only relevant stuff in that config is the chromeflags, which you can also pass to lighthouse CLI. Not that they helped in my case.
